acceptance/tests/pxp-module-puppet/negative/attempt_puppet_flag_whitelist_bypass.rb

```diff
@@ -0,0 +1,58 @@
+require 'pxp-agent/test_helper.rb'
+
+test_name 'C100245 - attempt bypass of flag whitelist with tab character' do
+
+  confine :except, :platform => ['windows']
+
+  step 'pick test host' do
+    @test_host = agents.first
+  end
+
+  step 'set up executable file' do
+    @executable_path = [@test_host.system_temp_path, 'run_me'].join('/')
+    @test_file = [@test_host.system_temp_path, (0...20).map { ('a'..'z').to_a[rand(26)] }.join].join('/')
+
+    executable_contents =<<-EOS
+echo 'ALL YOUR BASE BELONG TO US NOW' >> #{@test_file}
+    EOS
+
+    create_remote_file(@test_host, @executable_path, executable_contents)
+    on(@test_host, "chmod 777 #{@executable_path}")
+  end
+
+  teardown do
+    on(@test_host, "rm -f #{@executable_path}", :accept_all_exit_codes => true)
+    on(@test_host, "rm -f #{@test_file}", :accept_all_exit_codes => true)
+  end
+
+  step 'Ensure host has pxp-agent running and associated' do
+    on @test_host, puppet('resource service pxp-agent ensure=stopped')
+    create_remote_file(@test_host, pxp_agent_config_file(agent), pxp_config_json_using_puppet_certs(master, agent).to_s)
+    on @test_host, puppet('resource service pxp-agent ensure=running')
+
+    assert(is_associated?(master, "pcp://#{@test_host}/agent"),
+           "Agent #{@test_host} with PCP identity pcp://#{@test_host}/agent should be associated with pcp-broker")
+  end
+
+  step "Send an rpc_blocking_request to all agents" do
+    target_identities = []
+    target_identities << "pcp://#{@test_host}/agent"
+    begin
+      rpc_request(master, target_identities,
+                                       'pxp-module-puppet', 'run',
+                                       {:env => [], :flags => ['--noop',
+                                                               '--onetime',
+                                                               '--no-daemonize',
+                                                               "\t--postrun_command #{@executable_path}"]
+                                       })
+    rescue => exception
+      fail("Exception occurred when trying to run Puppet on all agents: #{exception.message}")
```

Comment:
This message could be more specific to this case (only one agent, and sending a deliberately dodgy flag).

Also a minor point - _fail_ can be confused with _fail_test_ , suggest changing it to either _fail_test_ or _raise_ depending on whether you intend a Failed or Errored test result (I think in this case you want _Errored_ )